contracts/multisig-prover/src/encoding/sol/mod.rs

```diff
@@ -0,0 +1,136 @@
+use std::{array::TryFromSliceError, collections::BTreeMap};
+
+use axelar_encoding::types::{
+    CrossChainId, Message, Payload, PublicKey, Signature, Signer, WeightedSignature, WorkerSet,
+    U256,
+};
+
+use itertools::Itertools;
+use multisig::{msg::SignerWithSig, verifier_set::VerifierSet};
+
+use crate::error::ContractError;
+
+const ECDSA_COMPRESSED_PUBKEY_LEN: usize = 33; // this should be probably a public constant in axelar_encoding.
+const ED25519_PUBKEY_LEN: usize = 32; // this should be probably a public constant in axelar_encoding.
+
+type Result<T> = core::result::Result<T, ContractError>;
+
+pub fn to_worker_set(vs: &VerifierSet) -> Result<WorkerSet> {
+    let mut signers: BTreeMap<String, Signer> = BTreeMap::new();
+
+    vs.signers
+        .iter()
+        .try_for_each(|(address, signer)| -> Result<()> {
+            let enc_signer = signer.address.to_string();
+            let enc_pubkey = to_pub_key(&signer.pub_key)?;
+
+            let enc_weight = U256::from_be(to_u256_be(signer.weight.u128()));
+
+            signers.insert(
+                address.clone(),
+                Signer::new(enc_signer, enc_pubkey, enc_weight),
+            );
+            Ok(())
+        })?;
+
+    Ok(WorkerSet::new(
+        vs.created_at,
+        signers,
+        U256::from_be(to_u256_be(vs.threshold.u128())),
+    ))
+}
+
+fn to_pub_key(pk: &multisig::key::PublicKey) -> Result<PublicKey> {
+    Ok(match pk {
+        multisig::key::PublicKey::Ecdsa(hb) => {
+            PublicKey::new_ecdsa(hb.to_array::<ECDSA_COMPRESSED_PUBKEY_LEN>()?)
+        }
+        multisig::key::PublicKey::Ed25519(hb) => {
+            PublicKey::new_ed25519(hb.to_array::<ED25519_PUBKEY_LEN>()?)
+        }
+    })
+}
+
+// Fits a u128 into a u256 in big endian representation.
+fn to_u256_be(u: u128) -> [u8; 32] {
+    let mut uin256 = [0u8; 32];
+    uin256[16..32].copy_from_slice(&u.to_be_bytes());
+    uin256
+}
```

Comment:
[see this](https://github.com/eigerco/solana-axelar-internal/issues/178#issuecomment-2160139501)

Reply:
fixed at 23be5851636afac76128f097f0353aa15bc3b7fe